### PR TITLE
[Mostly-Modular] Major refactor of most modular borg code with some QOL adjustments

### PR DESCRIPTION
**Pull request by Zergspower**
## About The Pull Request

Service borgs rejoice, you no longer have 3 duplicate items roundstart causing 5 whole-ass rows to be taken up. 

The majority of this PR is a refactor with a handful of tweaks to a couple of modules listed below, along with some locational standards for skins moving forward as the UI was causing some problems.

### Cargo borgs:
I drew blanks on what upgrades they can get moving forward, please feel free to suggest
- Boxcutter has been added
- Clamp **Upgrade** can now hold unwrapped crates, QOL mostly along with an increase in the storage capacity
-- basically it matches the modsuit carrying ability
___
### Mining borgs:
With all the fauna updates it's way past due to give them a little love
- Welder upgrade option to be researched
- PKA can fit 100 units from 80 to match actual PKA's 
___
### Service borgs: 
- Sanity returned to the loadout holy shit it was BAD
___
### Medical borgs: 
- Crew Pinpointer returned (Not the crew monitor)
___
### All borgs:
 - Sprite flexibility! You can now sprite a custom wire panel for borg skins

<details>
<Summary>How do?</Summary>

First grab the correct SKIN_ICON_STATE name 
  - e.g. engidarkhound for the Engineering Darkhound(best) borg skin
  
image
  - e.g. fmekaserv for the Service NikA(best) borg skin
  
image

1. ###  **_w** for open and wires exposed
  - e.g. engidarkhound_w for the darkhound(best) borg skin
  - e.g. fmekaserv_w for the NikA(best) borg skin

2. ### **_c** for cell removed
  - e.g. engidarkhound_c for the darkhound(best) borg skin
  - e.g. fmekaserv_c for the NikA(best) borg skin

3. ### **_cl** for the third state
  - e.g. engidarkhound_c for the darkhound(best) borg skin
  - e.g. fmekaserv_c for the NikA(best) borg skin

image

image

After you've made sure all three are made using the right **SKIN_ICON_STATE**, the lines above need the **R_TRAIT_UNIQUEPANEL** added to the **SKIN_FEATURES** List

image
</details>

- Standardized skin approach, it will go in order of 32x32  - > 64x32 - > 32x64 
-- we were having a lot of issues with the tallborgs being too swole and covering half of the 'next' window.

I also picked up the closed PR that went stale and incorporated it here, borgs currently cost way more than mechs and you can build **5!!** Ripleys from Print to moving to the bounty console before you can print and build 1 borg

https://github.com/Skyrat-SS13/Skyrat-tg/pull/23922

## How This Contributes To The Skyrat Roleplay Experience

More fun beeping?

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  _
  This took me over an hour validating that each skin worked as intended.
  
image
_
  Cargo borg cutely can cut packages open and the upgrade can hold four wrapped or unwrapped crates/packages
  
image
  
image
_
 New Service module layout (DEFAULT)

image
_
 Welder upgrade for Mining working as intended

image
_
Custom wire panels, one Darkhound was sacrificed in testing this. he'll never be able to get wired again

image
_
 Former loadout unfortunately caused a lot of overlapping, now you have to click next to get to the cute tall girls
 
image

  
</details>

## Changelog

:cl:Zergspower
add: Cargo borgs now gets a boxcutter! You can now manage packages properly
qol: Mining borg survival tweaks, Service borg overload tweaks
qol: Borgs can now have their own front panel sprite!
qol: Cargo borgs can now hold unwrapped crates with the upgraded clamp
qol: Reduces the cost and building time for cyborgs
balance: Mining borgs now have an upgraded welder option for research, the PKA nerf has been removed to match a normal one.
refactor: Most borg code has been refactored into one master module folder
/:cl:

### CODE REVIEW

**Comment by Exo-212** [comment]
Cargo borgs could be majorly improved with the ability to load crates added to their improved hydraulic clamp.

**Comment by Zergspower** [comment]
> Cargo borgs could be majorly improved with the ability to load crates added to their improved hydraulic clamp.

I was under the impression the clamp did that? Can you elaborate?

**Comment by Reco201** [comment]
> > Cargo borgs could be majorly improved with the ability to load crates added to their improved hydraulic clamp.
> 
> I was under the impression the clamp did that? Can you elaborate?

Only carries two crates, could be boosted to four imo

**Comment by Zergspower** [comment]
> > > Cargo borgs could be majorly improved with the ability to load crates added to their improved hydraulic clamp.
> > 
> > 
> > I was under the impression the clamp did that? Can you elaborate?
> 
> Only carries two crates, could be boosted to four imo

Sure, why not

**Comment by chemistrymain2** [comment]
> > > > Cargo borgs could be majorly improved with the ability to load crates added to their improved hydraulic clamp.
> > > 
> > > 
> > > I was under the impression the clamp did that? Can you elaborate?
> > 
> > 
> > Only carries two crates, could be boosted to four imo
> 
> Sure, why not

It is also designed to only lift wrapped crates, and can be made to lift unwrapped ones instead by including the path.

**Comment by Zergspower** [comment]
i dont play cargo, but the current clamp upgrade is kind of a sad excuse of an upgrade wow. Changing it to match the modsuit at least

**Comment by Zergspower** [comment]
image

**Comment by Zergspower** [comment]
upstream should make a bluespace mailbag for couriers so we can do the same here, like an RPED that just ZAPS it to the location you click

**Comment by Zergspower** [comment]
FYI the sprite tag is a false - we're just moving sprites in this PR

File: code/modules/mob/living/silicon/robot/robot_model.dm [hunk 1]
```diff
@@ -808,27 +808,23 @@
 		/obj/item/borg/apparatus/beaker/service,
 		/obj/item/reagent_containers/cup/beaker/large, //I know a shaker is more appropiate but this is for ease of identification
 		//Skyrat Edit Start: Borg Buff
-		//obj/item/reagent_containers/condiment/enzyme, //edit
-		/obj/item/reagent_containers/condiment/enzyme,
+		//obj/item/reagent_containers/condiment/enzyme, //edit - Borg shaker has it
+		/obj/item/borg/apparatus/beaker, // SR Edit: allows the pickup of different beakers for easier drink mixing
```
Comment (lessthnthree) [inline code/modules/mob/living/silicon/robot/robot_model.dm]:
```suggestion
		/obj/item/borg/apparatus/beaker, // SKYRAT EDIT: allows the pickup of different beakers for easier drink mixing
```

File: code/modules/mob/living/silicon/robot/robot_model.dm [hunk 2]
```diff
@@ -808,27 +808,23 @@
...
 		/obj/item/reagent_containers/dropper,
 		/obj/item/rsf,
 		/obj/item/storage/bag/tray,
+		/obj/item/storage/bag/tray,
```
Comment (lessthnthree) [inline code/modules/mob/living/silicon/robot/robot_model.dm]: Do you mean to have this twice?

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 3]
```diff
@@ -1183,7 +1186,7 @@
 	category = list(
 		RND_CATEGORY_MECHFAB_CYBORG_MODULES + RND_SUBCATEGORY_MECHFAB_CYBORG_MODULES_SERVICE
 	)
-	
+
```
Comment (lessthnthree) [inline code/modules/research/designs/mechfabricator_designs.dm:1189]: put whitespace back

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 4]
```diff
@@ -1,4 +1,6 @@
 //Cyborg
+
+/* //SKYRAT REMOVAL START - Costs lowered and time reduced -
```
Comment (lessthnthree) [inline code/modules/research/designs/mechfabricator_designs.dm]:
```suggestion
/* //SKYRAT EDIT REMOVAL BEGIN - Costs lowered and time reduced -
```

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 5]
```diff
@@ -75,6 +77,7 @@
 	category = list(
 		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
 	)
+*///SKYRAT REMOVAL END
```
Comment (lessthnthree) [inline code/modules/research/designs/mechfabricator_designs.dm]:
```suggestion
*///SKYRAT EDIT REMOVAL END
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 6]
```diff
@@ -61,3 +61,92 @@
 	construction_time = 120
 	category = list(RND_CATEGORY_MECHFAB_CYBORG_MODULES + RND_SUBCATEGORY_MECHFAB_CYBORG_MODULES_ALL)
 
+/datum/design/borg_upgrade_welding
+	name = "Welding Module"
+	id = "borg_upgrade_welding"
+	construction_time = 60
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 6 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 7]
```diff
@@ -61,3 +61,92 @@
...
+	category = list(RND_CATEGORY_MECHFAB_CYBORG_MODULES + RND_SUBCATEGORY_MECHFAB_CYBORG_MODULES_MINING)
+
+//Cyborg Skyrat overrides
+/datum/design/borg_suit
+	name = "Cyborg Endoskeleton"
+	id = "borg_suit"
+	build_type = MECHFAB
+	build_path = /obj/item/robot_suit
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT * 3)
+	construction_time = 80
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 8 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 8]
```diff
@@ -61,3 +61,92 @@
...
+		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
+	)
+
+/datum/design/borg_chest
+	name = "Cyborg Torso"
+	id = "borg_chest"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/chest/robot
+	materials = list(/datum/material/iron= SHEET_MATERIAL_AMOUNT * 8)
+	construction_time = 120
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 12 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 9]
```diff
@@ -61,3 +61,92 @@
...
+		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
+	)
+
+/datum/design/borg_head
+	name = "Cyborg Head"
+	id = "borg_head"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/head/robot
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT)
+	construction_time = 40
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 4 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 10]
```diff
@@ -61,3 +61,92 @@
...
+		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
+	)
+
+/datum/design/borg_l_arm
+	name = "Cyborg Left Arm"
+	id = "borg_l_arm"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/arm/left/robot
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT * 2)
+	construction_time = 80
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 8 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 11]
```diff
@@ -61,3 +61,92 @@
...
+		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
+	)
+
+/datum/design/borg_r_arm
+	name = "Cyborg Right Arm"
+	id = "borg_r_arm"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/arm/right/robot
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT * 2)
+	construction_time = 80
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 8 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 12]
```diff
@@ -61,3 +61,92 @@
...
+		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
+	)
+
+/datum/design/borg_l_leg
+	name = "Cyborg Left Leg"
+	id = "borg_l_leg"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/leg/left/robot
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT * 2)
+	construction_time = 80
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 8 SECONDS
```

File: modular_skyrat/modules/borgs/code/mechafabricator_designs.dm [hunk 13]
```diff
@@ -61,3 +61,92 @@
...
+		RND_CATEGORY_MECHFAB_CYBORG + RND_SUBCATEGORY_MECHFAB_CYBORG_CHASSIS
+	)
+
+/datum/design/borg_r_leg
+	name = "Cyborg Right Leg"
+	id = "borg_r_leg"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/leg/right/robot
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT * 2)
+	construction_time = 80
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/mechafabricator_designs.dm]:
```suggestion
	construction_time = 8 SECONDS
```

File: modular_skyrat/modules/borgs/code/robot_upgrade.dm [hunk 14]
```diff
@@ -242,3 +170,268 @@
...
+	unloading_time = 0.25 SECONDS
+	cooldown_duration = 0.25 SECONDS
+	whitelisted_item_types = list(/obj/item/mail)
+	whitelisted_item_description = "enveloppes"
+	item_weight_limit = WEIGHT_CLASS_NORMAL
+	clamp_sound_volume = 25
+	clamp_sound = 'sound/items/pshoom.ogg'
+
+/datum/design/borg_upgrade_clamp
+	name = "Improved Integrated Hydraulic Clamp Module"
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/robot_upgrade.dm]:
```suggestion
	name = "improved Integrated Hydraulic Clamp Module"
```

File: modular_skyrat/modules/borgs/code/robot_upgrade.dm [hunk 15]
```diff
@@ -242,3 +170,268 @@
...
+	name = "integrated rapid mail delivery device"
+	desc = "Allows you to carry around a lot of mail, to distribute it around the station like the good little mailbot you are!"
+	icon = 'icons/obj/service/library.dmi'
+	icon_state = "bookbag"
+	storage_capacity = 100
+	loading_time = 0.25 SECONDS
+	unloading_time = 0.25 SECONDS
+	cooldown_duration = 0.25 SECONDS
+	whitelisted_item_types = list(/obj/item/mail)
+	whitelisted_item_description = "enveloppes"
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/robot_upgrade.dm]: is this a typo

File: modular_skyrat/modules/borgs/code/robot_upgrade.dm [hunk 16]
```diff
@@ -242,3 +170,268 @@
...
+			to_chat(usr, span_warning("This unit already has a shrink module installed!"))
+			return FALSE
+		if(R_TRAIT_SMALL in borg.model.model_features)
+			to_chat(usr, span_warning("This unit's chassis cannot be shrunk any further."))
+			return FALSE
+
+		borg.hasShrunk = TRUE
+		ADD_TRAIT(borg, TRAIT_NO_TRANSFORM, REF(src))
+		var/prev_lockcharge = borg.lockcharge
+		borg.SetLockdown(1)
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/robot_upgrade.dm]:
```suggestion
		borg.SetLockdown(TRUE)
```

File: modular_skyrat/modules/borgs/code/robot_upgrade.dm [hunk 17]
```diff
@@ -242,3 +170,268 @@
...
+		borg.set_anchored(TRUE)
+		var/datum/effect_system/fluid_spread/smoke/smoke = new
+		smoke.set_up(1, location = get_turf(borg))
+		smoke.start()
+		sleep(0.2 SECONDS)
+		for(var/i in 1 to 4)
+			playsound(borg, pick('sound/items/drill_use.ogg', 'sound/items/jaws_cut.ogg', 'sound/items/jaws_pry.ogg', 'sound/items/welder.ogg', 'sound/items/ratchet.ogg'), 80, TRUE, -1)
+			sleep(1.2 SECONDS)
+		if(!prev_lockcharge)
+			borg.SetLockdown(0)
```
Comment (lessthnthree) [inline modular_skyrat/modules/borgs/code/robot_upgrade.dm]:
```suggestion
			borg.SetLockdown(FALSE)
```

**Review by lessthnthree — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/borgs/code/robot_upgrade.dm (hunk 15 above)
Comment (Zergspower) [reply modular_skyrat/modules/borgs/code/robot_upgrade.dm]: PFFFFF, i didnt spellcheck anything when i was moving other code

File: code/modules/mob/living/silicon/robot/robot_model.dm (hunk 2 above)
Comment (Zergspower) [reply code/modules/mob/living/silicon/robot/robot_model.dm]: It was requested by about four different people to leave the second tray, currently it was just for some reason in a different space
image

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 3 above)
Comment (Zergspower) [reply code/modules/research/designs/mechfabricator_designs.dm:1189]: I'm not really sure what this is, nothing else has more spaces here and i think VS did that change itself 
image

**Comment by Exo-212** [comment]
> image

Finally, cargo borgs are fully functional.

**Comment by desertmote** [comment]
I love the mining cyborg balance buffs, thank you. More QOL stuff would be:

1. a tool to butcher, we currently use the shovel and we lose this shovel if we upgrade our mining drill. meaning we do not take this upgrade ever
2. a point pass, preferably with a way to transfer points to a miner
3. potentially a way to defuse gibtonite

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 3 above)
Comment (lessthnthree) [reply code/modules/research/designs/mechfabricator_designs.dm:1189]: oh it's upstream, never mind

**Review by lessthnthree — APPROVED** [review]

**Comment by Zergspower** [comment]
> I love the mining cyborg balance buffs, thank you. More QOL stuff would be:
> 
> 1. a tool to butcher, we currently use the shovel and we lose this shovel if we upgrade our mining drill. meaning we do not take this upgrade ever
> 2. a point pass, preferably with a way to transfer points to a miner
> 3. potentially a way to defuse gibtonite

i did mean to add on the knife to the PKA, i'll do that in the next one

**Pull request merged**